manim/config.py

```diff
@@ -1,326 +1,430 @@
+                        'sound', 'leave_progress_bars']:
+        config[boolean_opt] = (default.getboolean(boolean_opt)
+                               if getattr(args, boolean_opt) is None
+                               else getattr(args, boolean_opt))
+    for str_opt in ['media_dir', 'video_dir', 'tex_dir',
+                    'text_dir']:
+        config[str_opt] = (default[str_opt]
+                           if getattr(args, str_opt) is None
+                           else getattr(args, str_opt))
+
+    # Handle the -t (--transparent) flag.  This flag determines which
+    # section to use from the .cfg file.
+    section = config_parser['transparent'] if args.transparent else default
+    for opt in ['png_mode', 'movie_file_extension',
+                'background_opacity']:
+        config[opt] = section[opt]
+
+    # Handle the -n flag.  Read first from the cfg and then override with CLI.
+    # These two are integers -- use getint()
+    for opt in ['from_animation_number', 'upto_animation_number']:
+        config[opt] = default.getint(opt)
+    if config['upto_animation_number'] == -1:
+        config['upto_animation_number'] = float('inf')
+    nflag = args.from_animation_number
+    if nflag is not None:
+        if ',' in nflag:
+            start, end = nflag.split(',')
+            config['from_animation_number'] = int(start)
+            config['upto_animation_number'] = int(end)
+        else:
+            config['from_animation_number'] = int(nflag)
+
+    # Handle the --dry_run flag.  This flag determines which section
+    # to use from the .cfg file.  All options involved are boolean.
+    section = config_parser['dry_run'] if args.dry_run else default
+    for opt in ['write_to_movie', 'write_all', 'save_last_frame', 'save_pngs', 'save_as_gif']:
+        config[opt] = section.getboolean(opt)
 
-__all__ = ["parse_cli", "get_configuration", "initialize_directories","register_tex_template","initialize_tex"]
+    # Read in the streaming section -- all values are strings
+    config['streaming'] = {opt: config_parser['streaming'][opt]
+                           for opt in ['live_stream_name', 'twitch_stream_key',
+                                       'streaming_protocol', 'streaming_ip',
+                                       'streaming_protocol', 'streaming_client',
+                                       'streaming_port', 'streaming_port',
+                                       'streaming_console_banner']}
 
+    # For internal use (no CLI flag)
+    config['skip_animations'] = any([config['save_last_frame'],
+                                     config['from_animation_number']])
 
-def parse_cli():
-    try:
-        parser = argparse.ArgumentParser()
+    return config
+
+
+def _parse_cli(arg_list, input=True):
+    parser = argparse.ArgumentParser(
+        description='Animation engine for explanatory math videos',
+        epilog='Made with <3 by the manim community devs'
```

Comment:
This can be a ❤, most people probably have unicode-capable terminals in 2020.

Reply:
I don't ;)